### PR TITLE
LibWeb: Update Location initialization according to spec

### PR DESCRIPTION
**Pull request by shlyakpavel**

### CODE REVIEW

File: Libraries/LibWeb/HTML/Location.cpp [hunk 1]
```diff
@@ -38,13 +38,35 @@ void Location::visit_edges(Cell::Visitor& visitor)
...
 {
     Base::initialize(realm);
     WEB_SET_PROTOTYPE_FOR_INTERFACE(Location);
 
-    // FIXME: Implement steps 2.-4.
+    auto& vm = this->vm();
 
+    // Step 2: Let valueOf be location's relevant realm.[[Intrinsics]].[[%Object.prototype.valueOf%]].
+    auto& intrinsics = realm.intrinsics();
+    auto value_of_function = MUST(intrinsics.object_prototype()->get(vm.names.valueOf));
```
Comment (yyny) [inline Libraries/LibWeb/HTML/Location.cpp:51]: Other code accesses intrinsics/prototypes uses `->get_without_side_effects(...)`, so you could probably use that here too.

**Comment by shlyakpavel** [comment]
As mentioned by @yyny , I should test this with
https://wpt.fyi/results/html/browsers/history/the-location-interface/location-symbol-toprimitive.html?label=master&label=experimental&product=ladybird&aligned&q=location
https://wpt.fyi/results/html/browsers/history/the-location-interface/location-valueof.html?label=master&label=experimental&product=ladybird&aligned&q=location

**Comment by shlyakpavel** [comment]
Well, we do
> 5. Set the value of the [[DefaultProperties]] internal slot of location to location.[[OwnPropertyKeys]]().

This adds toPrimitive to DefaultProperties 
As per spec
When, as per spec, we do
>  2. If the value of the [[DefaultProperties]] internal slot of this contains P, then set desc.[[Configurable]] to true.`

This finds toPrimitive in DefaultProperties and sets it's Configurable to true
When, wpt test checks
`assert_false(prop.configurable)`
And fails. Is this a spec issue? WPT issue? My reading of the spec issue?

**Comment by shlyakpavel** [comment]
Related: https://github.com/whatwg/html/issues/4157

I propose not to care about that WPT for now and merge this "as is", as this corresponds to what spec wants us to do. If there's anything to modify, it's `Location::internal_get_own_property` anyway, and that is not part of this issue.

**Comment by gmta** [comment]
@shlyakpavel 

> I propose not to care about that WPT for now

What do other browsers do with the configurability here?

**Comment by shlyakpavel** [comment]
@gmta as far as I understand, other browsers do not respect [location-getownproperty](https://html.spec.whatwg.org/multipage/history.html#location-getownproperty) spec. That function and its implementation are outside this PR scope at the moment.

**Pull request merged**